automates/text_reading/src/main/scala/org/clulab/aske/automates/apps/ExtractAndAlign.scala

```diff
@@ -550,58 +566,174 @@ object ExtractAndAlign {
+      dstLinkElement = dstElements(alignment.dst)
+      score = alignment.score
+    } yield mkHypothesis(srcLinkElement, dstLinkElement, score, debug)
+  }
+
+  def mkLinkHypothesisFromValueSequences(srcElements: Seq[Value], dstElements: Seq[Value], alignments: Seq[Seq[Alignment]], debug: Boolean): Seq[Obj] = {
+
+    //    println("len src el1 " + srcElements.length)
+    //    println("len dst el2 " + dstElements.length)
     for {
       topK <- alignments
       alignment <- topK
       srcLinkElement = srcElements(alignment.src)
       dstLinkElement = dstElements(alignment.dst)
       score = alignment.score
-    } yield mkHypothesis(srcLinkElement, dstLinkElement, score)
+    } yield mkHypothesis(srcLinkElement, dstLinkElement, score, debug)
   }
 
 
-  def getLinkHypotheses(linkElements: Map[String, Seq[String]], alignments: Map[String, Seq[Seq[Alignment]]]): Seq[Obj] = {//, SVOGroungings: Map[String, Seq[sparqlResult]]): Seq[Obj] = {
+
+  def getLinkHypotheses(linkElements: Map[String, Seq[String]], alignments: Map[String, Seq[Seq[Alignment]]], debug: Boolean): Seq[Obj] = {//, SVOGroungings: Map[String, Seq[sparqlResult]]): Seq[Obj] = {
     // Store them all here
     val hypotheses = new ArrayBuffer[ujson.Obj]()
     val linkElKeys = linkElements.keys.toSeq
     // Comment -> Text
     if (linkElKeys.contains(COMMENT) && linkElKeys.contains(TEXT_VAR)) {
-      hypotheses.appendAll(mkLinkHypothesis(linkElements(COMMENT), linkElements(TEXT_VAR), alignments(COMMENT_TO_TEXT)))
+      hypotheses.appendAll(mkLinkHypothesis(linkElements(COMMENT), linkElements(TEXT_VAR), alignments(COMMENT_TO_TEXT), debug))
     }
 
 
     // Src Variable -> Comment
     if (linkElKeys.contains(SOURCE) && linkElKeys.contains(COMMENT)) {
       println("has source and comment")
-      hypotheses.appendAll(mkLinkHypothesis(linkElements(SOURCE), linkElements(COMMENT), alignments(SRC_TO_COMMENT)))
+      hypotheses.appendAll(mkLinkHypothesis(linkElements(SOURCE), linkElements(COMMENT), alignments(SRC_TO_COMMENT), debug))
     }
 
 
     // Equation -> Text
     if (linkElKeys.contains(EQUATION) && linkElKeys.contains(TEXT_VAR)) {
       println("has eq and text")
-      hypotheses.appendAll(mkLinkHypothesis(linkElements(EQUATION), linkElements(TEXT_VAR), alignments(EQN_TO_TEXT)))
+      hypotheses.appendAll(mkLinkHypothesis(linkElements(EQUATION), linkElements(TEXT_VAR), alignments(EQN_TO_TEXT), debug))
     }
 
     // TextVar -> TextDef (text_span)
     //taken care of while creating link elements
     if (linkElKeys.contains(TEXT_VAR) && linkElKeys.toSeq.contains(TEXT)) {
-      hypotheses.appendAll(mkLinkHypothesisTextVarDef(linkElements(TEXT_VAR), linkElements(TEXT)))
+      hypotheses.appendAll(mkLinkHypothesisTextVarDef(linkElements(TEXT_VAR), linkElements(TEXT), debug))
     }
 
     hypotheses
   }
 
 
+  def get2PaperLinkHypotheses(linkElements: Map[String, Seq[String]], alignment: Seq[Seq[Alignment]], debug: Boolean): Seq[Obj] = {
```

Comment:
@maxaalexeeva I'm seeing a lot of methods named `get2Paper...()` can we rename these methods using the schema `getInterPaper...()`.

For instance `get2PaperLinkHypotheses()` would become `getInterPaperLinkHypotheses()`. I prefer this naming schema because 2-paper is a little ambiguous. 2-paper could refer to getting link hypotheses from two separate papers, but in this case we are talking about making link hypotheses among two papers (hence the prefix _inter_).

Reply:
@pauldhein, good point! But paper1LinkElements and paper2LinkElements-like names are okay?

Reply:
@maxaalexeeva yes those names are fine since they reference the specific objects of a paper specified by an index.